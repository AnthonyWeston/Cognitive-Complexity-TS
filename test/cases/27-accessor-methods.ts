class Getter {
    // depth 0
    get f(): string {
        // depth 0
        return this.f; // +1 for recursion
    }
}

class Setter {
    set f(value) {
        this.f = 1;
    }
}
