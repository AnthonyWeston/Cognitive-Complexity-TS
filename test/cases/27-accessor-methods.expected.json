{
    "27-accessor-methods.ts": {
        "score": 2,
        "inner": [
            {
                "name": "Getter",
                "score": 1,
                "line": 1,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 3,
                        "column": 5
                    }
                ]
            },
            {
                "name": "Setter",
                "score": 1,
                "line": 9,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 10,
                        "column": 5
                    }
                ]
            }
        ]
    }
}